app/src/app/api/v0/user/[user]/file/route.ts

```diff
@@ -56,20 +46,11 @@ export const POST = apiHandler(
       throw new createHttpError.Unauthorized("Unauthorized");
     }
 
-    const user = await db.models.User.findOne({
-      attributes: ["userId"],
-      where: {
-        userId: userId,
-      },
-    });
-
-    if (!user) {
-      throw new createHttpError.NotFound("User not allowed");
-    }
-
     const formData = await req.formData();
```

Comment:
Should we validate formData using zod as well? You're not doing null checks below for some of the properties, so it might make sense and give better error messages than the internal server error that is the result when the user passes the wrong data types in some fields here.